tools/github-event-processor/Azure.Sdk.Tools.GitHubEventProcessor/GitHubEventClient.cs

```diff
@@ -279,10 +282,37 @@ public async Task WriteRateLimits(string prependMessage = null)
         }
 
         /// <summary>
-        /// Write the current rate limit and remaining number of transactions.
+        /// Return the number of updates a scheduled task can make. The Core Rate Limit that GitHub Actions can make is 15000/hour
+        /// for enterprise and 1000/hour for non-enterprise. The max number of results that can be retried from SearchIssues is 1000.
+        /// The CoreRateLimit is set when WriteRateLimits is called and this is done at the start of processing in Main. If the core
+        /// rate limit is 15000, return 1000, otherwise return 100 which is 1/10th of the hourly limit for non-enterprise repository.
         /// </summary>
-        /// <param name="prependMessage">Optional message to prepend to the rate limit message.</param>
-        public async Task WriteSearchRateLimits(string prependMessage = null)
+        /// <returns>The number updates a scheduled task can make.</returns>
+        public virtual async Task<int> ComputeScheduledTaskUpdateLimit()
+        {
+            // CoreRateLimit will be set in WriteRateLimits but if that hasn't been called yet, call it now.
+            if (CoreRateLimit == 0)
+            {
+                var miscRateLimit = await GetRateLimits();
+                CoreRateLimit = miscRateLimit.Resources.Core.Limit;
+            }
+            // If the repository is an enterprise repository allow the max number search results, 1000, to be processed.
+            if (CoreRateLimit == RateLimitConstants.ActionRateLimitEnterprise)
```

Comment:
This exact check makes me a little uneasy, but I guess we don't have any other great way to figure this out. 

Should we instead always do `CoreRateLimit/10` and max it at 1000?

Reply:
I was thinking if we wanted to do a greater than/less than vs. exact and choose it dynamically, but at the same time I think their rate limits are rather predictable.

Reply:
Sure can, it'll be updated with the next commit.